api/src/pcapi/core/object_storage/backends/local.py

```diff
@@ -48,6 +49,23 @@ def delete_public_object(self, folder: str, object_id: str) -> None:
         try:
             os.remove(file_local_path)
             os.remove(str(file_local_path) + ".type")
+        except FileNotFoundError:
+            # replicate gpc backend behavior on file not found
+            logger.info("File not found on deletion on local storage: %s", file_local_path)
         except OSError as exc:
             logger.exception("An error has occured while trying to delete file on local file storage: %s", exc)
             raise exc
+
+    def list_files(self, folder: str, *, max_results: int = 1000) -> list[str]:
+        results = []
+        breaking = False
+        local_dir_len = len(str(self.local_dir("", ""))) + 1
+        for root, dirs, files in os.walk(self.local_dir(folder, "")):
+            for item in chain(dirs, files):
+                results.append(f"{root}/{item}"[local_dir_len:])
+                if len(results) >= max_results:
+                    breaking = True
+                    break
+            if breaking:
+                break
```

Comment:
Sans variable : 
```suggestion
                else:
                    continue
                break
```

Ou une version compréhensible : 
```suggestion
                    return results
```